src/plugins/home/server/services/sample_data/data_sets/util.ts

```diff
@@ -68,17 +82,29 @@ export const getSavedObjectsWithDataSource = (
 
       if (dataSourceTitle) {
         if (
-          saveObject.type === 'dashboard' ||
-          saveObject.type === 'visualization' ||
-          saveObject.type === 'search'
+          savedObject.type === 'dashboard' ||
+          savedObject.type === 'visualization' ||
+          savedObject.type === 'search'
         ) {
-          saveObject.attributes.title = saveObject.attributes.title + `_${dataSourceTitle}`;
+          savedObject.attributes.title = savedObject.attributes.title + `_${dataSourceTitle}`;
         }
       }
 
-      return saveObject;
+      return savedObject;
     });
   }
 
-  return saveObjectList;
+  return savedObjectList;
+};
+
+export const getWorkspaceIntegratedSavedObjects = (
+  savedObjectList: SavedObject[],
+  workspaceId?: string
+) => {
+  savedObjectList = cloneDeep(savedObjectList);
```

Comment:
I am a little bit confused here, could we assign the copied object to a new variable?

Reply:
There are too many attributes update work in `overrideSavedObjectId`. It's all based references, it will take huge effects if we want to use new variables. It

Reply:
I guess @SuZhou-Joe want to say: shall we do `const savedObjectListCopy = cloneDeep(savedObjectList);`?